envoy/network/dns.h

```diff
@@ -66,7 +66,15 @@ class DnsResponse {
   DnsResponse(const std::string& host, uint16_t port, uint16_t priority, uint16_t weight)
       : response_(SrvResponse{host, port, priority, weight}) {}
 
-  const AddrInfoResponse& addrInfo() const { return absl::get<AddrInfoResponse>(response_); }
+  const AddrInfoResponse addrInfo() const {
```

Comment:
@zuercher  changed the signature to return `value` instead of reference for proper thread synchronisation. Apart from memory consumption, it shouldn't add more complexity.

TBH, I am not completely sure this is what we should do or if there are other ways to do this. One thing probably could be to Assert() in this case instead. Please let me know your thoughts

Reply:
We can move this logic to [#line65](https://github.com/envoyproxy/envoy/pull/29231/files#diff-6e80985af9d2f1d49e11a4dbaf343fa54286f6767d4ac449620d37b949ac5d9eR65). There should no `Assert()` in this Envoy's data plane as a general guidance.

Reply:
Thanks @botengyao, done.